automotive/maliput/utility/generate_obj.cc

```diff
@@ -158,6 +189,257 @@ class GeoFace {
   std::vector<GeoNormal> normals_;
 };
 
+// @return If two adjacent quads on R can be merged into a single one that will
+// represent the same surface. The received faces are expected to be quads
+// (4-vertex faces), share a side of the quad and vertices to be defined
```

Comment:
@basicNew what do you mean by `share a side`? Isn't this function checking if they do?

Reply:
Also, thinking out loud, I see assertions for vertex count but not for vertex order (I recall a cross product, torque-like approach a friend of ours implemented to determine something similar).

Reply:
Yes, totally, that is part of what the function checks.

About cross product, torque-like approach I think it is an overkill. If we were to do that sort of checks then all the file should be rewritten.

Reply:
Meh, you skipped all the fun stuff :)

Reply:
hahaha